src/components/DataTable.vue

```diff
@@ -0,0 +1,85 @@
+<template>
+  <el-table :data="tableData" height="88vh" @cell-click="handleCellClick">
+    <el-table-column
+      v-for="col in tableColumns"
+      :key="col.id"
+      :prop="col.prop"
+      :label="col.label"
+      :sortable="col.isSortable"
+      :column-key="col.prop"
+      :width="col.width"
+    />
+    <el-table-column prop="action" label="">
+      <template #default="table">
+        <div class="action-icons">
+          <i
+            class="fa-solid fa-pencil icon"
+            @click="
+              (event) => {
+                $emit('edit', table.row);
+                openAppModal();
+                preventRowExpansion(event);
+              }
+            "
+          ></i>
+          <i
+            class="fa-solid fa-trash icon"
+            @click="
+              (event) => {
+                $emit('delete', table.row);
+                preventRowExpansion(event);
+              }
+            "
+          ></i>
+        </div>
+      </template>
+    </el-table-column>
+  </el-table>
+  <el-pagination
+    layout="prev, pager, next"
+    :total="tableData.length"
+    @current-change="(newPageNumber) => $emit('pageChange', newPageNumber)"
+  >
+  </el-pagination>
+</template>
+
+<script>
+import { mapMutations } from 'vuex';
+import { OPEN_APP_MODAL } from '@/store/mutation-types';
+import { ElTable, ElTableColumn, ElPagination } from 'element-plus';
+
+export default {
+  name: 'DataTable',
+  components: {
+    ElTable,
+    ElTableColumn,
+    ElPagination,
+  },
+  props: {
+    tableData: {
+      type: Array,
+      required: true,
+    },
+    tableColumns: {
+      type: Array,
+      required: true,
+    },
+  },
+  emits: ['edit', 'delete', 'pageChange'],
+  methods: {
+    ...mapMutations({ openAppModal: OPEN_APP_MODAL }),
+    handleCellClick: (row, column, cell) => {
```

Comment:
Vue method should never be an arrow function, try logging `this` inside of it
```suggestion
    handleCellClick: (row, column, cell) => {
```

Reply:
I figured it out. Fixed.